constrainttemplates/peerauthn-mesh-strict-mtls.yaml

```
@@ -0,0 +1,33 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  annotations:
+    description: Enforce the mesh level strict mtls PeerAuthentication.
+  name: peerauthnmeshstrictmtls
+spec:
+  crd:
+    spec:
+      names:
+        kind: PeerAuthnMeshStrictMtls
+      validation:
+        openAPIV3Schema:
+          type: object
+          properties:
+            rootNamespace:
+              description: Istio root namespace, default value is "istio-system" if not specified.
+              type: string
+  targets:
+  - target: admission.k8s.gatekeeper.sh
+    rego: |-
+      package istio.security.peerauthentication
+
+      violation[{"msg": msg}] {
+        root_ns := object.get(object.get(input, "parameters", {}), "rootNamespace", "istio-system")
```

need to check if the namespace from the input matches `root_ns`:

```suggestion
        root_ns := object.get(object.get(input, "parameters", {}), "rootNamespace", "istio-system")
        input.review.object.metadata.name == root_ns
```